src/main/java/com/dsrm/dsrmbackend/services/impl/RoomTypeServiceImpl.java

```diff
@@ -1,34 +1,51 @@
 package com.dsrm.dsrmbackend.services.impl;
 
 import com.dsrm.dsrmbackend.dto.RoomTypeRequestDTO;
+import com.dsrm.dsrmbackend.entities.Room;
 import com.dsrm.dsrmbackend.entities.RoomType;
 import com.dsrm.dsrmbackend.mappers.RoomTypeMapper;
+import com.dsrm.dsrmbackend.repositories.RoomRepo;
 import com.dsrm.dsrmbackend.repositories.RoomTypeRepo;
 import com.dsrm.dsrmbackend.services.RoomTypeService;
 import lombok.RequiredArgsConstructor;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
 
 @Service
 @RequiredArgsConstructor
 public class RoomTypeServiceImpl implements RoomTypeService {
-    private final RoomTypeRepo repository;
+    private final RoomTypeRepo roomTypeRepo;
+    private final RoomRepo roomRepo;
     private final RoomTypeMapper roomTypeMapper;
     public RoomType addRoomType(RoomTypeRequestDTO roomTypeReqDto) {
         RoomType roomType = roomTypeMapper.roomTypeReqDTOToRoomType(roomTypeReqDto);
         roomType.setId(UUID.randomUUID().toString());
-        return repository.save(roomType);
+        return roomTypeRepo.save(roomType);
     }
 
     public Optional<RoomType> getRoomType(String roomId) {
-        return repository.findById(roomId);
+        return roomTypeRepo.findById(roomId);
     }
 
     public Page<RoomType> getRoomTypes(Pageable pageable) {
-        return repository.findAll(pageable);
+        return roomTypeRepo.findAll(pageable);
+    }
+
+    @Override
+    public Optional<RoomType> deleteRoomType(String id) {
```

Comment:
We want to use cascade delete like here https://www.javatpoint.com/jpa-cascade-remove

Reply:
Added CascadeType.REMOVE but it seems it does not set null in Room entity but it removes it too